sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatViewModel.kt

```diff
@@ -262,10 +272,18 @@ abstract class ChatViewModel<ARGS: NavArgs>(
 
             withContext(default) {
                 for (message in messages) {
+                    val messageColor = Color.parseColor(
+                        userColorsHelper.getHexCodeForKey(
+                            message.getColorKey(),
+                            app.getRandomHexCode()
+                        )
+                    )
```

Comment:
we should be setting the color key id to the `MessageHolderViewState` and querying the helper upon bind instead of doing the work here such that it can be lazily set to the view when the data is bound to the holder.

This also keeps the helper object out of the viewmodel (and it's inheritors) such that we can scope it properly for just that instance of the Fragment or Adapter.